### PR TITLE
Add .editorconfig file

### PR DESCRIPTION
**Pull request by newbazz**
It specifies the indentation of
a file irrespective of the text
editor or IDE.

Closes https://github.com/coala/coala/issues/4097

**For short term contributors:** we understand that getting your commits well
defined like we require is a hard task and takes some learning. If you
look to help without wanting to contribute long term there's no need
for you to learn this. Just drop us a message and we'll take care of brushing
up your stuff for merge!

### Checklist

- [x] I read the [commit guidelines](http://coala.io/commit) and I've followed
      them.
- [x] I ran coala over my code locally. (*All commits have to pass
      individually.* It is not sufficient to have "fixup commits" on your PR,
      our bot will still report the issues for the previous commit.) You will
      likely receive a lot of bot comments and build failures if coala does not
      pass on every single commit!

After you submit your pull request, **DO NOT click the 'Update Branch' button.**
When asked for a rebase, consult [coala.io/rebase](https://coala.io/rebase)
instead.

Please consider helping us by reviewing other peoples pull requests as well:

- pick up any PR at <https://coala.io/review>
- review it (check <https://coala.io/reviewing> for more info)
- if you are sure that it needs work, use `cobot mark wip <URL>` to get it out
  of the review queue.

The more you review, the more your score will grow at coala.io and we will
review your PRs faster!

### CODE REVIEW

**Comment by newbazz** [comment]
@RaiVaibhav ok so what sud I do next, or it will be done by someone else?

**Comment by RaiVaibhav** [comment]
@newbazz  you have to do nothing i just approved your PR and added my comment on that

**Comment by newbazz** [comment]
@RaiVaibhav okay thank you.

**Comment by kartikeys98** [comment]
ack 5c61455

**Comment by Makman2** [comment]
@jayvdb could you take a look if this is enough and what you've expected? :)

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+charset = utf-8
+indent_style = space
+indent_size = 4
+trim_trailing_whitespace = true
+
+# 2 space indentation for yml fies and package.json
+[{package.json,*.yml}]
+indent_style = space
+indent_size = 2
+trim_trailing_whitespace = true
```
Comment (mohitkyadav) [inline .editorconfig]: Please insert a new line at the end.

Comment (newbazz) [reply .editorconfig]: Can you help me out coz I have mistakenly deleted the branch. :P

Comment (newbazz) [reply .editorconfig]: I actually have the branch present in local.

Comment (mohitkyadav) [reply .editorconfig]: Then just push your local branch to origin.

Comment (newbazz) [reply .editorconfig]: Sry not in local I have it in origin.

Comment (mohitkyadav) [reply .editorconfig]: Then pull it.

Comment (newbazz) [reply .editorconfig]: Done thanks.

Comment (newbazz) [reply .editorconfig]: Changes made.

File: .editorconfig [hunk 2]
```diff
@@ -0,0 +1,19 @@
+root = true
+
+[**]
+end_of_line = lf
```
Comment (aashutoshrathi) [inline .editorconfig]: Add `charset = utf-8` here rather than python block

Comment (newbazz) [reply .editorconfig]: sry but did not get u

Comment (newbazz) [reply .editorconfig]: instead of [**] should i write charset = utf-8 ?

File: .editorconfig (hunk 1 above)
Comment (mohitkyadav) [reply .editorconfig]: :+1:

File: .editorconfig (hunk 2 above)
Comment (mohitkyadav) [reply .editorconfig]: No, @newbazz Add a line `charset = utf-8`.

Comment (newbazz) [reply .editorconfig]: @mohitkyadav okk thanks.

Comment (newbazz) [reply .editorconfig]: I made the changes.

File: .editorconfig [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+# Set default charset and indentation for python files
+[{*.py, coalib/**/*.py, ./coala, tests/**/*.py}]
+charset = utf-8
+indent_style = space
+indent_size = 4
+trim_trailing_whitespace = true
+
+# 2 space indentation for yml fies and package.json
+[{package.json,*.yml}]
```
Comment (mohitkyadav) [inline .editorconfig]: Please add a space after`,`

Comment (newbazz) [reply .editorconfig]: Is this the reason for build fail?

Comment (mohitkyadav) [reply .editorconfig]: No. That's unknown for me.

Comment (newbazz) [reply .editorconfig]: Ok, btw thats not the reason mostly! :P

**Comment by jayvdb** [comment]
@gitmate-bot rebase

File: .editorconfig [hunk 4]
```diff
@@ -0,0 +1,20 @@
+root = true
+
+[**]
+charset = utf-8
+end_of_line = lf
+insert_final_newline = true
+max_line_length = 80
+
+# Set default charset and indentation for python files
+[{*.py, coalib/**/*.py, ./coala, tests/**/*.py}]
```
Comment (mohitkyadav) [inline .editorconfig]: @newbazz take a look at the samples on http://editorconfig.org/
And remove spaces after `,` and sorry for misguiding you earlier😅

**Comment by newbazz** [comment]
Is travis still down?

**Comment by newbazz** [comment]
@jayvdb which review is not addressed?

**Comment by aashutoshrathi** [comment]
@newbazz remove the `,` in line 10.

File: .editorconfig (hunk 3 above)
Comment (p8107406332) [reply .editorconfig]: hey can i take this issue

**Comment by newbazz** [comment]
Please review

**Comment by mohitkyadav** [comment]
ack 65b3c5b

**Comment by newbazz** [comment]
Theres a pr on the same issue by someone else!!

File: .editorconfig [hunk 5]
```diff
@@ -0,0 +1,20 @@
+root = true
+
+[**]
+charset = utf-8
+end_of_line = lf
+insert_final_newline = true
+max_line_length = 80
+
+# Set default charset and indentation for python files
+[{*.py,coalib/**/*.py,./coala, tests/**/*.py}]
```
Comment (jayvdb) [inline .editorconfig]: remove the space here

**Comment by jayvdb** [comment]
@newbazz where is the other PR?  Use links please.

**Comment by jayvdb** [comment]
unack 65b3c5b

**Comment by newbazz** [comment]
@jayvdb made the changes, and the pr i as talking abt is https://github.com/coala/coala/pull/4679.

**Comment by jayvdb** [comment]
Please rebase your branch.

**Comment by newbazz** [comment]
@jayvdb can you plz tell me how? git pull rebase?

**Comment by jayvdb** [comment]
Read the docs.

**Comment by newbazz** [comment]
@jayvdb rebased

**Comment by newbazz** [comment]
Please review.

File: .editorconfig [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+
+[**]
+charset = utf-8
+end_of_line = lf
+insert_final_newline = true
+max_line_length = 80
+
+# Set default charset and indentation for python files
+[{*.py,coalib/**/*.py,./coala,tests/**/*.py}]
+charset = utf-8
```
Comment (aashutoshrathi) [inline .editorconfig]: If we have `charset = utf-8` in `[**]` block, then no need to add it in `[{*.py,coalib/**/*.py,./coala,tests/**/*.py}]` block.

**Comment by newbazz** [comment]
@aashutoshrathi changes made.

**Comment by newbazz** [comment]
Please review.

**Comment by aashutoshrathi** [comment]
Please squash the three commits to one:
First, sync your master with coala master and then rebase.
```
git checkout master
git pull upstream/master
git push upstream/master
git checkout file
git rebase -i master
```

**Comment by RaiVaibhav** [comment]
@newbazz  :+1:  but just need a little squash thing and IMO after squash this commit, for future please use `soft reset` and `force push to avoid squashing things`  for your next commits on `same PR`,It seriously helps me a lot.

**Comment by newbazz** [comment]
@RaiVaibhav u say i sud use git reset --soft HEAD~3

**Comment by newbazz** [comment]
@RaiVaibhav @aashutoshrathi  done!!

**Comment by aashutoshrathi** [comment]
@newbazz Resolve conflicts

**Comment by RaiVaibhav** [comment]
@newbazz  I don't know about your log, I generally use `reset --soft`, please don't get confuse use what suited you best

**Comment by newbazz** [comment]
@RaiVaibhav I resolved the conflict but now how to solve this?

**Comment by aashutoshrathi** [comment]
@newbazz just remove redundant line 155.

**Comment by newbazz** [comment]
@aashutoshrathi still soe error is coming

**Comment by aashutoshrathi** [comment]
@newbazz, please rebase.
Always rebase your commits, never make it even 2.

**Comment by jayvdb** [comment]
An edit conflict, and 4 commits.  i.e. a cluster f*(&^

**Comment by newbazz** [comment]
@jayvdb  resolving that.

**Comment by newbazz** [comment]
Why is the pr not reopening?

**Comment by newbazz** [comment]
I will resolve this pr by tonight. sorry for the inconvinence.

**Comment by adtac** [comment]
You have deleted the branch associated with this PR. That's why you can't reopen the PR.

**Comment by newbazz** [comment]
Yeah it was done by mistake.
Am trying to resolve that only.

**Comment by newbazz** [comment]
I reopened the PR but it created a new one. Anyone help me why?

**Pull request closed without merging**